File: src/485/solution.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int curr_consec, max_consec;
        curr_consec = 0;
        max_consec = 0;
        
        for(int i = 0; i < nums.size(); ++i){
            if(nums[i] == 1) {
                curr_consec = curr_consec + 1;
            }
            else {
                if(max_consec < curr_consec){
                    max_consec = curr_consec;
                }
                curr_consec = 0;
            }
        }
        
        if(max_consec >= curr_consec){
            return max_consec;
        }
        return curr_consec;
    }
};

File: src/485/solution.py
from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr_max = 0
        curr = 0
        for num in nums:
            if num:
                curr += 1
            else:
                curr_max = max(curr, curr_max)
                curr = 0
        
        return max(curr, curr_max)